source/slang/slang-lower-to-ir.cpp

```diff
@@ -5170,7 +5170,11 @@ struct DeclLoweringVisitor : DeclVisitor<DeclLoweringVisitor, LoweredValInfo>
                         // Need to construct a sub-witness-table
                         auto irWitnessTableBaseType = lowerType(subContext, astReqWitnessTable->baseType);
                         irSatisfyingWitnessTable = subBuilder->createWitnessTable(irWitnessTableBaseType);
-
+                        auto mangledName = getMangledNameForConformanceWitness(
+                            subContext->astBuilder,
+                            astReqWitnessTable->witnessedType,
+                            astReqWitnessTable->baseType);
```

Comment:
This isn't relevant to this PR, but we actually have a problem in Slang today in terms of how we represent the mangled names of conformance witnesses. Right now we generate the mangled name for a witness in a way that doesn't care where it came from; all that matters is the base type and the derived type.

That's actually technically incorrect, and can cause problems if two modules introduce the same conformance via `extension`s.

If the module that declares the conformance is the same as the one that declared the concrete type, or the interface, the current naming is reasonable.

If the module that declares the conformance is distinct from the ones that declare the concrete type and the interface, we really need the mangled name to include the module name that introduced the conformance. We could always apply this rule if we want to keep things simple.

The problem with including the module name, of course, is that it breaks the logic added to the public API here, where we assume that the mangled name for a conformance only depends on the types involved.